### PR TITLE
Feature/tests

### PR DESCRIPTION
**Pull request by lgiordani**
This is an implementation of #91 and #92 and partially #93.

### CODE REVIEW

**Comment by lgiordani** [comment]
The first commit is an error, sorry, the added file has been already removed.

File: tox.ini [hunk 1]
```diff
@@ -4,10 +4,11 @@ envlist = py34, py33, py32, py27, py27-configparser, pypy
 [testenv]
 passenv = HOME
 deps=
+  distribute
```
Comment (peritus) [inline tox.ini]: Is that needed ?

File: tox.ini [hunk 2]
```diff
@@ -4,10 +4,11 @@ envlist = py34, py33, py32, py27, py27-configparser, pypy
...
   pytest
   mock
 commands=
-  py.test [] tests.py
+  py.test [] tests.py tests
```
Comment (peritus) [inline tox.ini]: We should probably move tests.py to the tests folder (and maybe call them end-to-end or cli tests or something) instead.

**Comment by peritus** [comment]
I really like this, seems way clearer to me now.

Also a great step to maybe one day have a documented api that is not the cli, but a python api that can be used by other programs.

File: tox.ini (hunk 1 above)
Comment (lgiordani) [inline tox.ini]: This comes from an error I get on my Ubuntu 14.10 when running tox. It complains that 

ImportError: No module named 'setuptools'

I solved it by explicitly installing it in the virtualenv with this line. This error comes from the fact that I have no setuptools installed on my machine, probably.

File: tox.ini (hunk 2 above)
Comment (lgiordani) [inline tox.ini]: A good idea. Do you prefer I commit a change to the branch or submit another PR?

Comment (peritus) [inline tox.ini]: The smaller and the more focused on one aspect the better (a rule I live by for both for commits/pull requests). Don't worry about form, I'll split it up as needed. Me just needing to click the merge button might get the code in faster though.

**Comment by lgiordani** [comment]
Do you need other changes here? Otherwise I'll start working on some next steps.

**Comment by peritus** [comment]
Sorry, just time constraints on my side. I hope to have a few hours to look at this in detail soon.

I like the general direction and hope to get it integrated sooner than later.

**Comment by lgiordani** [comment]
Sorry, I didn't want to hurry you, no problem, I will start from here to
try and split some other parts of the whole program.
Take your time ;)

Leonardo Giordani
@tw_lgiordani <http://twitter.com/tw_lgiordani> - lgiordani.com
My profile on About.me <http://about.me/leonardo.giordani> - My GitHub page
<https://github.com/lgiordani>

**Comment by peritus** [comment]
No worries :)

**Comment by lgiordani** [comment]
I opened #98 to clean up a bit of mess with Git history, so that new PR do not contain wrong commits.
